Makefile

```diff
@@ -5,16 +5,48 @@ include $(shell test -f .tardigrade-ci || curl -sSL -o .tardigrade-ci "https://r
 
 pytest/install:
 	@ $(MAKE) install/pip/$(@D) PYPI_PKG_NAME=$(@D)
-	@ python -m pip install -r lambda/tests/requirements_dev.txt
+	@ python -m pip install \
+		-r lambda/tests/requirements_dev.txt \
+		-r tests/requirements_test.txt
 
+.PHONY: python/deps
 python/deps:
 	@ echo "[$@] Installing package dependencies"
 	@ python -m pip install -r lambda/src/requirements.txt
 
+.PHONY: python/test
 python/test: | guard/program/pytest
 python/test:
 	@ echo "[$@] Starting Python tests"
-	pytest
+	pytest lambda/tests
 	@ echo "[$@]: Tests executed!"
 
-.PHONY: python/deps python/test
+.PHONY: terraform/pytest
+terraform/pytest: | guard/program/terraform guard/program/pytest
+	@ echo "[$@] Starting test of Terraform lambda installation"
+	@ echo "[$@] Terraform 'apply' command is slow ... be patient !!!"
+	pytest tests
+	@ echo "[$@]: Completed successfully!"
+
+.PHONY: localstack/pytest localstack/up localstack/down localstack/clean
+localstack/pytest: | guard/program/terraform guard/program/pytest
+	@ echo "[$@] Running Terraform tests against LocalStack"
+	DOCKER_RUN_FLAGS="--network host --rm" \
+		$(MAKE) docker/run target=terraform/pytest
+	@ echo "[$@]: Completed successfully!"
+
+localstack/up: | guard/program/terraform guard/program/pytest
+	@ echo "[$@] Starting LocalStack"
+	docker-compose -f tests/docker-compose-localstack.yml up --detach
+
+localstack/down: | guard/program/terraform guard/program/pytest
+	@ echo "[$@] Stopping and removing LocalStack container"
+	docker-compose -f tests/docker-compose-localstack.yml down --rm all
+
+localstack/clean:
+	@ echo "[$@] Stopping and removing LocalStack container and images"
+	@ $(MAKE) localstack/down
+	docker images | grep lambci | awk '{print $$1 ":" $$2}' | \
```

Comment:
Done; it seems a bit gimmicky.  I have an alternative solution for the golang/terratest version (not checked in yet).

Reply:
Actually, that change fails:
```[localstack/clean] Stopping and removing LocalStack container and images
set +o pipefail; docker images | grep lambci | \
	awk '{print $1 ":" $2}' | xargs docker rmi
"docker rmi" requires at least 1 argument.
See 'docker rmi --help'.

Usage:  docker rmi [OPTIONS] IMAGE [IMAGE...]

Remove one or more images
make: *** [localstack/clean] Error 123
```

Reply:
Missed a bit. Add `-r` to xargs. This will fail because xargs will invoke the command even when stdin is empty:

```
set +o pipefail; docker images | grep lambci | \
	awk '{print $1 ":" $2}' | xargs docker rmi
```

Using `-r` with xargs will not invoke the command when stdin is empty:

```
set +o pipefail; docker images | grep -E "lambci|new-account-iam-role" | awk '{print $1 ":" $2}' | \
        xargs -r docker rmi
```

Reply:
I missed the -r after args and the fact that you combined the two.  It's ugly.

Reply:
Yeah, I liked the idea of combining them because 1) extra shells and duplicative logic can be expensive, and 2) it opens the possibility of exposing an env to let the user extend the behavior, e.g. `TARDIGRADE_DOCKER_CLEAN_IMAGES`, and having the tardigrade-ci target `docker/clean` do the work (that target is broke anyway and needs something like this so it doesn't purge the entire cache).

Reply:
I did not combine them.

Reply:
It's already a complicated and over-engineered statement.